src/core/reducers/addons.js

```diff
@@ -287,7 +289,14 @@ export default function addonsReducer(
       const { addons } = action.payload;
       const newState = { ...state };
       Object.keys(addons).forEach((key) => {
-        newState[key] = createInternalAddon(addons[key]);
+        const addon = createInternalAddon(addons[key]);
+
+        // We index add-ons by id and slug to be able to retrieve them by any
+        // of these parameters. This is needed to redirect an add-on detail
+        // page loaded by ID.
+        // See: https://github.com/mozilla/addons-frontend/issues/342
+        newState[addon.id] = addon;
```

Comment:
Could you file an issue to stop storing `addon` in state twice and link it to a TODO comment here? It will be an annoying patch to write but we should eventually store the `addon` only once and reference it by ID in all other parts of the state.

Reply:
hmm isnt it this one? https://github.com/mozilla/addons-frontend/issues/3421

Reply:
That issue is different but related. I think it would be fine to edit the description of that issue to make it work for both :)

Reply:
indeed!